### PR TITLE
chore(C): add doubly linked list

### PR DESCRIPTION
**Pull request by Ashborn-SM**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] New algorithm
- [ ] Optimization in previous algorithms
- [ ] Code style update
- [ ] Refactor
- [ ] Documentation
- [ ] Other, please describe:

**Briefly describe the changes in this PR**
add doubly linked list using C

**Other information:**

### CODE REVIEW

**Comment by Ashborn-SM** [comment]
@UG-SEP waiting for your review

File: algorithms/C/linked-lists/doubly-linked-list.c [hunk 1]
```diff
@@ -0,0 +1,133 @@
+#include <stdio.h>
+#include <stdlib.h>
+
```
Comment (aayushjain7) [inline algorithms/C/linked-lists/doubly-linked-list.c:3]: provide aim/description for the program

File: algorithms/C/linked-lists/doubly-linked-list.c [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+	pop_front(list);
+	pop_back(list);
+
+	print(list);
+
+	delete(list);
+
+	return 0;
+}
```
Comment (aayushjain7) [inline algorithms/C/linked-lists/doubly-linked-list.c:133]: also, provide 1-2 test cases with time and space complexity

**Review by aayushjain7 — CHANGES_REQUESTED** [review]

**Review by aayushjain7 — APPROVED** [review]
great work!

**Pull request merged**